spec/javascripts/models/beta/story_spec.js

```diff
@@ -375,4 +375,123 @@ describe('Story model', function () {
       expect(changedStory.errors).toEqual([error]);
     });
   });
+
+  describe('isNew', () => {
+    it('returns true when story id is null', () => {
+      const story = { id: null }
+
+      expect(Story.isNew(story)).toBe(true);
+    });
+
+    it("returns false when story id isn't null", () => {
+      const story = { id: 42 }
+
+      expect(Story.isNew(story)).toBe(false);
+    });
+  });
+
+  describe('canDelete', () => {
+    it("returns true when story isn't accepted and isn't new", () => {
+      const story = { id: 42, state: 'started' };
+
+      expect(Story.canDelete(story)).toBe(true);
+    });
+
+    it('returns false when story is accepted', () => {
+      const story = { id: 42, state: 'accepted' };
+
+      expect(Story.canDelete(story)).toBe(false);
+    });
+
+    it('returns false when story is new', () => {
+      const story = { id: null, state: 'started' };
+
+      expect(Story.canDelete(story)).toBe(false);
+    });
+  });
+
+  describe('canSave', () => {
+    it("returns true when story isn't accepted and the text isn't empty", () => {
+      const story = { state: 'started', _editing: { title: 'title' } };
+
+      expect(Story.canSave(story)).toBe(true);
+    });
+
+    it('returns false when story is accepted', () => {
+      const story = { state: 'accepted', _editing: { title: 'title' } };
+
+      expect(Story.canSave(story)).toBe(false);
+    });
+
+    it('returns false when story title is null', () => {
+      const story = { state: 'started', _editing: { title: "" } };
+
+      expect(Story.canSave(story)).toBe(false);
+    });
+  });
+
+  describe('removeEmptyStory', () => {
+    it('remove a story with null id from a stories array', () => {
+      const stories = [{ id: 1 }, { id: 2 }];
+      const emptyStory = { id: null };
+
+      const storiesWithEmptyStories = [
+        stories[0],
+        emptyStory,
+        stories[1]
+      ];
+
+      expect(Story.removeEmptyStory([...stories, emptyStory])).toEqual(stories);
+      expect(Story.removeEmptyStory([emptyStory, ...stories])).toEqual(stories);
+      expect(Story.removeEmptyStory(storiesWithEmptyStories)).toEqual(stories);
+    });
+  });
+
+  describe('createNewStory', () => {
+    const stories = [{ id: 1 }, { id: 3 }]
+
+    it('returns a empty story', () => {
+      const story = Story.createNewStory(stories, {});
+
+      expect(story.id).toBe(null);
+    });
+
+    it('returns a expanded story', () => {
+      const story = Story.createNewStory(stories, {});
+
+      expect(story.collapsed).toBe(false);
+    });
+  });
+
+  describe('replaceEmptyOrAddNewStory', () => {
+    it('replace an empty for a new story', () => {
```

Comment:
What do you think about creating a test that asserts that it'll reuse the attributes from the empty one?

Also, a general idea for this PR: we should stop using the word "empty" since it may not be empty 🤔, what do you think about using "not persisted" instead?